BUILDER_DEV.md

```diff
@@ -1,390 +1,294 @@
 
-`config_sessionsrv.toml`
-```toml
-[permissions]
-admin_team = 1995301
-build_worker_teams = [2555389]
-early_access_teams = [1995301]
 ```
+$ sudo make build-srv
+```
+
+* Open up the builder-worker file
 
-`config_worker.toml`
-```toml
+```
+$ sudo vim /hab/svc/builder-worker
+```
+
+* Add your Github Auth Key:
+
+**/hab/svc/builder-worker/config.toml**
+```
+auth_token = ""
+```
+
+Should become:
+
+**/hab/svc/builder-worker/config.toml**
+```
 auth_token = "<your github token>"
-bldr_url = "http://localhost:9636"
-auto_publish = true
-airlock_enabled = false
 ```
 
+* Run this command (it may take awhile) to start all the Builder services
+
+```
+$ sudo -E make bldr-run
+```
+
+### UI Setup
+
+* Open a separate shell on your VM
+* Install npm
+
+```
+$ sudo apt-get install npm
+```
+
+* Cd into the builder-web component
+
+```
+$ cd path/to/your/habitat/repo/components/builder-web
+```
+
+* Install all dependencies
+
+```
+$ npm install
+```
+
+* Create the habitat.conf.js file
+
+```
+$ cp habitat.conf.sample.js habitat.conf.js
+```
+
+* Start the UI
+
+```
+$ sudo npm start
+```
+
+* Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
+
+### Creating an Origin
 
-(Note: If you want your log files to persist across restarts of your development machine, replace `/tmp` with some other directory. It *must* exist and be writable before you start the job server).
+* Navigate to http://localhost:3000/#/pkgs
+* Log in
+* Click on "Create New Origin"
+* Fill out form (call the origin "core"), click "Save & Continue"
+* Click on "Keys"
+* Download the public and private key
+* Copy the public key contents
+* From a terminal, run
 
-Now, modify the `Procfile` (located in your hab repo in the `support` folder) to point the api, sessionsrv, jobsrv and worker services to the previously created config files, e.g.
+```
+$ hab origin key import
+```
+
+* Paste the public key contents
+* Hit Ctrl + D twice to close the stream
+* Run this again:
 
 ```
-api: target/debug/bldr-api start --config /home/your_alias/habitat/config_api.toml
-sessionsrv: target/debug/bldr-sessionsrv start --config /home/your_alias/habitat/config_sessionsrv.toml
-worker: target/debug/bldr-worker start --config /home/your_alias/habitat/config_worker.toml
-jobsrv: target/debug/bldr-jobsrv start --config /home/your_alias/habitat/config_jobsrv.toml
+$ hab origin key import
 ```
 
-## Build the Builder services for the first time
-1. Open a new terminal window.
-1. Run `make build-srv`
+* Paste the private key contents
+* Hit Ctrl + D twice to close the stream
 
-Note: this only needs to be done the first time to get the pieces in place. If you clean the project, make sure to unset the variables show below when you do a clean build, otherwise the build will look for packages on your local build service!
+### Install Dependencies in your local Builder Env
```

Comment:
I'm not sure all 4 of these `hab origin key import`s are necessary. Installing the `bldr.habitat.sh` public key definitely is (or else `.hart` files `hab install`ed from `bldr.habitat.sh` couldn't be `hab package upload`ed), but I'm not sure about the other 3.

Reply:
I confirmed that I can successfully build packages with only the one key installed.